monkestation/code/modules/nightmare/nightmare_helpers.dm

```diff
@@ -0,0 +1,50 @@
+#define PASSIVE_SNUFF_LIMIT 0.5
+
+/// Handles checking for nearby weak lights to auto-snuff as a nightmare. Start turf arg is optional and can be any atom, it'll just have get_turf called on it.
+/proc/check_passive_nightmare_snuff(mob/living/carbon/nightmare, turf/start)
+	if(!istype(nightmare))
+		return
+
+	if(!istype(start))
+		start = get_turf(start || nightmare)
+
+	if(!istype(start)) // Really? How did you even... whatever, just ignore it.
+		return
+
+	var/lumcount = start.get_lumcount()
+	if(lumcount < SHADOW_SPECIES_LIGHT_THRESHOLD || lumcount > PASSIVE_SNUFF_LIMIT)
+		return
+
+	var/obj/item/organ/internal/heart/nightmare/heart = nightmare.get_organ_slot(ORGAN_SLOT_HEART)
+
+	if(!istype(heart) || !heart.blade)
+		return
+
+	var/snuffed_something = FALSE
+
+	for(var/atom/nearby as anything in view(2, start))
```

Comment:
instead of doing it this way you should use a proximity_monitor

Reply:
How would that work when it "skips" a tile? Specifically, due to how phasing works, it checks around the tile you're trying to move to instead of the one you're currently on.

Reply:
NO. NO PROXIMITY MONITOR. NO MORE.